multicluster/service-mirror/cluster_watcher.go

```diff
@@ -837,11 +848,24 @@ func (rcsw *RemoteClusterServiceWatcher) Start(ctx context.Context) error {
 					return
 				}
 
-				if !isExportedEndpoints(old, rcsw.log) {
+				epOld, ok := old.(*corev1.Endpoints)
+				if !ok {
+					rcsw.log.Errorf("error processing endpoints object: got %#v, expected *corev1.Endpoints", epOld)
 					return
 				}
 
-				rcsw.eventsQueue.Add(&OnUpdateEndpointsCalled{new.(*corev1.Endpoints)})
+				epNew, ok := new.(*corev1.Endpoints)
+				if !ok {
+					rcsw.log.Errorf("error processing endpoints object: got %#v, expected *corev1.Endpoints", epNew)
+					return
+				}
+
+				if !isExportedEndpoints(epOld) && !isExportedEndpoints(epNew) {
```

Comment:
How does this work when the export label is removed?  In that case, the epOld would have the exported label but the epNew would not and processing would be skipped.  Actually, I don't see a codepath for unexporting endpoints anywhere.

Reply:
Indeed, I opened #8475 to track that separately.